src/session/components/avatar.rs

```diff
@@ -111,74 +124,127 @@ impl Avatar {
 
     fn setup_expressions(&self) {
         let item_expression = Self::this_expression("item");
+        let imp = self.imp();
 
-        let icon_name_expression = item_expression.chain_property::<AvatarItem>("icon-name");
-
-        let custom_image_expression = item_expression.chain_property::<AvatarItem>("image");
-        let custom_image_expression =
-            gtk::ClosureExpression::new::<Option<gtk::gdk::Paintable>, _, _>(
-                &[
-                    icon_name_expression.clone().upcast(),
-                    custom_image_expression.upcast(),
-                ],
-                closure!(|_: Self,
-                          icon_name: Option<String>,
-                          cusom_image: Option<gtk::gdk::Paintable>| {
-                    match icon_name {
-                        Some(_) => None,
-                        None => cusom_image,
-                    }
-                }),
-            );
-
-        let show_initials_expression = icon_name_expression.chain_closure::<bool>(closure!(
-            |_: Self, icon_name: Option<String>| icon_name.is_none()
-        ));
-
-        let text_expression = item_expression.chain_property::<AvatarItem>("display-name");
-        let text_expression = gtk::ClosureExpression::new::<Option<String>, _, _>(
-            &[
-                icon_name_expression.clone().upcast(),
-                text_expression.upcast(),
-            ],
-            closure!(
-                |_: Self, icon_name: Option<String>, display_name: Option<String>| {
-                    icon_name
-                        // If we use an icon for the avatar, we always want its color to be blue. As
-                        // AdwAvatar doesn't allow us to set the color in an explicit manner, we are
-                        // forced to use this workaround.
-                        .map(|_| Some(String::from("-")))
-                        .unwrap_or(display_name)
+        // Chat title expression
+        let title_expression = item_expression.chain_property::<Chat>("title");
+        gtk::ClosureExpression::new::<String, _, _>(
+            &[item_expression.clone(), title_expression],
+            closure!(|_: Self, chat: Chat, title: String| {
```

Comment:
This doesn't seem to work. The avatar is still not blue on my side.
This on the other side does work (Can't make a proper suggestion here)
```rust
closure!(|_: Self, item: Option<glib::Object>, title: String| {
    item.as_ref()
        .and_then(|i| i.downcast_ref::<Chat>())
        .map(Chat::is_own_chat)
        .map(|_| "-".to_string())
        .unwrap_or(title)
}),
```

Reply:
Now I understand why you did it differently compared to line 149. 
I am not totally against it, but I think relying on such unstable conditions is a potential source of error.
As the code grows, such things are very likely to be forgotten.

Reply:
> I am not totally against it, but I think relying on such unstable conditions is a potential source of error.
As the code grows, such things are very likely to be forgotten.

Fair. Honestly the problem here is that expression are a bit inconsistent, like PropertyExpression can just downcast to the correct type if they can and they safely ignore the expression if they can't, but ClosureExpressions just makes the program panic if the type cannot cast to the correct one.

Reply:
> Now as I understand the PropertyExpression acts like guard!? If there is something other than a Chat the ClosureExpression will not be executed?!

Yeah, exactly. That's how PropertyExpressions work. They "chain" with the other expression only when the object in that expression exists and it can cast correctly to the wanted type. But that's not true for ClosureExpressions. But if you have a ClosureExpression + PropertyExpressions this works, honestly I'm not too sure why.

Reply:
Mh, actually it works because the ClosureExpression executes only when every expressions chained to it are all valid, so yeah, in this case the PropertyExpression acts like a guard because of that.